app/code/Magento/Sales/Model/Order/Config.php

```diff
@@ -211,7 +218,7 @@ public function getStateStatuses($state, $addLabels = true)
                 foreach ($collection as $item) {
                     $status = $item->getData('status');
                     if ($addLabels) {
-                        $statuses[$status] = $item->getStoreLabel();
+                        $statuses[$status] = $area == 'adminhtml' ? $item->getLabel() : $item->getStoreLabel();
```

Comment:
Great looks like this fixes the problem. It does appear that this is the same check as in `getStatusLabel`, assuming that `$item` here and `$status` in the `getStatusLabel` are the same type. Is it possible to use the same method for getting the label based on the area?